src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -55,10 +61,21 @@ client_ptr kafka_client_cache::fetch_or_insert(
     if (it_hash == inner_hash.end()) {
         if (_cache.size() >= _cache_max_size) {
             if (!_cache.empty()) {
+                using namespace std::chrono_literals;
+
                 auto item = inner_list.back();
                 vlog(plog.debug, "Cache size reached, evicting {}", item.key);
                 inner_list.pop_back();
                 _evicted_items.push_back(std::move(item));
+
+                // Trigger the eviction process when the timer is set to fire in
+                // the next 5s
+                auto window = ss::lowres_clock::now() + 5s;
+                if (
+                  _clean_and_evict_timer.armed()
+                  && _clean_and_evict_timer.get_timeout() < window) {
+                    _clean_and_evict_timer.rearm(ss::lowres_clock::now());
+                }
```

Comment:
i'm not certain about this debouncing logic. it seems like it should be something more like

1. if the timer isn't armed, then we do want to arm it so it will run. but let's arm it for a few seconds from now
2. if the timer is armed, and it is going to run soon, then do nothing
3. if the timer is armed, and it is going to run too far into the future, then reschedule it for a few seconds from now

is this capturing such debounce logic? maybe i'm missing it.

Reply:
Yeah my logic is off. Thanks for the notes! I'll change in the next push